File: Javascript/recusive_binary_search.js
function recursive_binary_search(list, target) {
  console.log(list);
  if (list.length < 1) {
    return false;
  } else {
    const midpoint = Math.floor(list.length / 2);
    if (list[midpoint] === target) {
      return true;
    } else {
      if (list[midpoint] < target) {
        recursive_binary_search(list.splice(midpoint + 1, list.length), target);
      } else {
        recursive_binary_search(list.splice(0, midpoint), target);
      }
    }
  }
  return false;
}

function verify(result) {
  console.log("Item found: ", result);
}

verify(recursive_binary_search([1, 2, 3, 4, 5, 6, 7, 8, 9, 10], 12));
verify(recursive_binary_search([1, 2, 3, 4, 5, 6, 7, 8, 9, 10], 6));

File: python/recusive_binary_search.py
def recursive_binary_search(List, target):
    if len(List) == 0:
        return False

    else:
        midpoint = len(List)//2
        if List[midpoint] == target:
            return True
        else:
            if List[midpoint] < target:
                recursive_binary_search(List[midpoint+1:], target)
            else:
                recursive_binary_search(List[:midpoint], target)
    return False

def verify(result):
    print("Item found:", result)


numbers = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]

result = recursive_binary_search(numbers, 12)
verify(result)

result = recursive_binary_search(numbers, 6)
verify(result)
